### ISSUE TITLE
Not works

### ISSUE THREAD

**Issue by AlexSheff** [Gourieff/sd-webui-roop-nsfw#3]
The original roop working - no issues, On this version I get Error 
Error running postprocess_batch: D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py
Traceback (most recent call last):
  File "D:\AI_text\stable-diffusion-webui\modules\scripts.py", line 496, in postprocess_image
    script.postprocess_image(p, pp, *script_args)
  File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 212, in postprocess_image
    upscale_options=self.upscale_options,
  File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 133, in upscale_options
    return UpscaleOptions(
TypeError: UpscaleOptions.__init__() got an unexpected keyword argument 'do_restore_first'

**Comment by convex-mind** [comment]
That's exactly the problem I'm having too. Same error message.

**Comment by Gourieff** [comment]
Hm... What settings did you guys set to catch this error?

**Comment by Gourieff** [comment]
I tried different scenarios, but didn't get this error
Did you restart Automatic1111 after update the extension?

**Comment by AlexSheff** [comment]
> Hm... What settings did you guys set to catch this error?

Did you restart Automatic1111 after update the extension? 
- Yes I did, even restarted PC. 
- Still have Error Total progress: 100%|██████████████████████████████████████████████████████████████████| 20/20 [00:12<00:00,  1.63it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:12<00:00,  1.60it/s]
Error running postprocess_batch: D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py/s]
Traceback (most recent call last):
  File "D:\AI_text\stable-diffusion-webui\modules\scripts.py", line 496, in postprocess_image
    script.postprocess_image(p, pp, *script_args)
  File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 212, in postprocess_image
    upscale_options=self.upscale_options,
  File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 133, in upscale_options
    return UpscaleOptions(
TypeError: UpscaleOptions.__init__() got an unexpected keyword argument 'do_restore_first'
[Screenshot](https://prnt.sc/kCxvdWKtVjnX) 
Any suggestions? ))

**Comment by convex-mind** [comment]
Standard settings. Nothing special. 
It's enabled, comma seperated face options (for above and target) are set on 0
Also checked with restore face selected and unselected.
A complete restart is already done. Still the same issue.
Standard roop works without errors.

**Comment by Gourieff** [comment]
![bug](https://github.com/Gourieff/sd-webui-roop-nsfw/assets/85128026/bd613955-2817-4cd1-a360-a31888940369)
You have an old version of extension (0.0.2) for some reason
Current version is 0.1.0
Try to go to `\extensions\sd-webui-roop-nsfw` folder inside sd-webui directory
Run Terminal or Console (cmd or powershell) for this folder
And update the extension manually by `git pull`

**Comment by Gourieff** [comment]
If it doesn't help (I'm sure it might) - try to delete the extension and install it once again

**Comment by Gourieff** [comment]
The best way to update extensions properly in Automatic1111 - is to to do it outside of webgui by `git pull` them
Otherwise, sometimes there can be some update issues

**Comment by convex-mind** [comment]
The reinstallation changed my version of 0.0.3 to 0.1.0
Also did a "git pull" on the \extensions\sd-webui-roop-nsfw folder. However everthing is "Already up to date".
Restarted everything and still have the same errors in the exact lines like AlexSheff posted.
Nonetheless, thank you for your efforts, Gourieff! I hope it can be solved somehow. (At the moment i can switch back to the 0.0.3 version).

**Comment by AlexSheff** [comment]
Even after the update Still have
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:27<00:00, 1.38s/it]
Error running postprocess_batch: D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py/s]
Traceback (most recent call last):
File "D:\AI_text\stable-diffusion-webui\modules\scripts.py", line 496, in postprocess_image
script.postprocess_image(p, pp, *script_args)
File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 212, in postprocess_image
upscale_options=self.upscale_options,
File "D:\AI_text\stable-diffusion-webui\extensions\sd-webui-roop-nsfw\scripts\faceswap.py", line 133, in upscale_options
return UpscaleOptions(
TypeError: UpscaleOptions.init() got an unexpected keyword argument 'do_restore_first'
and in WebUI shown old version 0.0.2

**Comment by Gourieff** [comment]
Really nonetheless 😵
I tried to reinstall it (through webui) to an old clear copy of Automatic1111 (Jan2023 version), and even there it works ok and shows the right version number
Idk guys how to help ¯\_(ツ)_/¯ very strange bug
![prntscrn](https://github.com/Gourieff/sd-webui-roop-nsfw/assets/85128026/33d860f9-ffe1-4ba9-8e15-3d27f258fb76)

**Comment by Gourieff** [comment]
With the version 0.0.2 there was another caption of roop accordion tab - smth like "nsfw-roop v0.0.2" (it takes version number from another variable), and caption was changed to "NSFW-Roop" with version 0.1.0
It seems that some files got the newest update and some file are still in old commits for some strange reason

Try guys to extract the whole `scripts` folder from the original archive https://github.com/Gourieff/sd-webui-roop-nsfw/archive/refs/heads/main.zip
Delete the current `scripts` folder inside the `\extensions\sd-webui-roop-nsfw` directory and copy the extracted one there

It MUST help

**Comment by convex-mind** [comment]
I implemented it as you described. Unfortunately it didn't help.

That's why I've now deleted everything that had to do with roop. This also includes the standard installation of Roop, which I had at the same time.

I reinstalled your new version and now everything is working fine.
As a result I think that your script is error free and instead something must have gotten messed up before in conjunction with the other roop installation and its updates.

What caused that now, I can't say. But I'm used to these kinds of problems from Stable Diffusion and its numerous extensions. You have to try a lot and sometimes do something completely new.

Thanks for your patience and your help!

**Comment by AlexSheff** [comment]
I have **delete original version roop** and 

> extract the whole scripts folder from the original archive https://github.com/Gourieff/sd-webui-roop-nsfw/archive/refs/heads/main.zip
Delete the current scripts folder inside the \extensions\sd-webui-roop-nsfw directory and copy the extracted one there
Now it works. 
Thank you!

**Comment by Connum** [comment]
This still seems to be an issue. I just installed this extension feshly fom the URL to this repository. I encountered that error and indeed, the version that got installed was 0.0.2 and it didn't find a newer version. Something must be wrong!

**Comment by AlexSheff** [comment]
> This still seems to be an issue. I just installed this extension feshly fom the URL to this repository. I encountered that error and indeed, the version that got installed was 0.0.2 and it didn't find a newer version. Something must be wrong!

Delete folders all roop from extensions and reinstall new version from URL

**Comment by Connum** [comment]
That seems to have worked, thanks! Strange that there was a problem in the first place...

**Comment by Icarushollow** [comment]
i've done every single thing people have said worked on this thread as well as the instructions at first. I tried the initial instructions: install from URL ---> restart UI; didnt work (btw when installing, it is instant. the console just says: "Checking ReActor (ex-roop-GE) requirements... Ok" -- and nothing changes and Auto1111 says it's installed the moment that it starts)
I then tried to disable roop and restart UI - didnt work
I then tried to delete the roop files/folder - didn't work
I then unzipped the linked archive in the ReActor folder 'scripts' - didn't work
I tried to then reinstall roop and deactivate it on the UI - didn't work
so what am I to try now? It's unclear what I'm supposed to do at this point. I've never had an extension have such a complicated setup and troubleshoot. Just click, install, done.


Error running postprocess_batch: C:\Users\Joe\stable-diffusion-webui\extensions\sd-webui-reactor\scripts\faceswap.py
Traceback (most recent call last):
  File "C:\Users\Joe\stable-diffusion-webui\modules\scripts.py", line 454, in postprocess_image
    script.postprocess_image(p, pp, *script_args)
  File "C:\Users\Joe\stable-diffusion-webui\extensions\sd-webui-reactor\scripts\faceswap.py", line 235, in postprocess_image
    upscale_options=self.upscale_options,
  File "C:\Users\Joe\stable-diffusion-webui\extensions\sd-webui-reactor\scripts\faceswap.py", line 151, in upscale_options
    return UpscaleOptions(
TypeError: UpscaleOptions.__init__() got an unexpected keyword argument 'do_restore_first'
